src/openms/source/FEATUREFINDER/MassTraceDetection.cpp

```
@@ -251,312 +257,452 @@
+                next_up_peak_int = spec_trace_up[next_up_peak_idx_left].getIntensity();
+                next_up_peak_im = spec_trace_up.getFloatDataArrays()[Ion_Mobility_idx][next_up_peak_idx_left];
+
+                if (next_up_peak_idx_right != next_up_peak_idx_left)
+                {
+                  std::map<int, double> ionMobilityMap;
+                  for (Size im_i = next_up_peak_idx_left; im_i <= next_up_peak_idx_right; ++im_i)
+                  {
+                    ionMobilityMap[im_i] = spec_trace_up.getFloatDataArrays()[Ion_Mobility_idx][im_i];
+                  }
+
+                  double closestValue = ionMobilityMap.begin()->second;
```

Cppcheck (2.10-2)

[error] Out of bounds access in expression 'ionMobilityMap.begin()' because 'ionMobilityMap' is empty.

(containerOutOfBounds)